tests/test_odilo.py

```diff
@@ -0,0 +1,250 @@
+# encoding: utf-8
+import os
+import json
+import unittest
+
+from flask import (
+    Flask,
+)
+
+from api.odilo import (
+    OdiloAPI,
+    MockOdiloAPI,
+    OdiloCirculationMonitor,
+    FullOdiloCollectionMonitor,
+    RecentOdiloCollectionMonitor,
+)
+
+from api.circulation import (
+    CirculationAPI,
+    CirculationException
+)
+
+from sqlalchemy.ext.declarative import declarative_base
+from flask_sqlalchemy_session import flask_scoped_session
+from api.config import Configuration
+
+from . import (
+    DatabaseTest,
+    sample_data
+)
+
+from core.model import (
+    Collection,
+    ConfigurationSetting,
+    DataSource,
+    DeliveryMechanism,
+    ExternalIntegration,
+    Identifier,
+    Library,
+    LicensePool,
+    Representation,
+    RightsStatus,
+    SessionManager
+)
+
+Base = declarative_base()
+app = Flask(__name__)
+
+testing = True
+# testing = 'TESTING' in os.environ
+db_url = Configuration.database_url(testing)
+# Initialize a new database session unless we were told not to
+# (e.g. because a script already initialized it).
+autoinitialize = os.environ.get('AUTOINITIALIZE') != 'False'
+if autoinitialize:
+    SessionManager.initialize(db_url)
+session_factory = SessionManager.sessionmaker(db_url)
+_db = flask_scoped_session(session_factory, app)
+if autoinitialize:
+    SessionManager.initialize_data(_db)
```

Comment:
You should not need to set up a fake Flask application or do any of this stuff. Everything you need to test should be manageable with a regular `DatabaseTest`.